crates/bevy_ecs/src/storage/table.rs

```diff
@@ -699,7 +732,7 @@ mod tests {
     fn table() {
         let mut components = Components::default();
         let mut storages = Storages::default();
-        let component_id = components.init_component::<W<usize>>(&mut storages);
+        let component_id = components.init_component::<W<u32>>(&mut storages);
```

Comment:
```suggestion
        let component_id = components.init_component::<W<TableRow>>(&mut storages);
```
it is incredibly sketchy to be inserting data of type `T` as a component init'd as type `U`. We should really get around to adding debug asserts for this at some point...

Reply:
Fixed.